cpu/stm32_common/periph/i2c_2.c

```diff
@@ -117,13 +119,11 @@ void i2c_init(i2c_t dev)
 #endif
 
     /* configure device */
-    DEBUG("[i2c] init: configuring device\n");
     _i2c_init(i2c, i2c_config[dev].clk, ccr);
 
 #if defined(CPU_FAM_STM32F4)
     /* make sure the analog filters don't hang -> see errata sheet 2.14.7 */
     if (i2c->SR2 & I2C_SR2_BUSY) {
-        DEBUG("[i2c] init: line busy after reset, toggle pins now\n");
```

Comment:
This is because if you do enable debug it doesn't actually work and seems to screw up the shell, I think it has something to do with the uart being initialized after the i2c but I didn't really look into it.   I wrote something above the enable debug to indicate that.  It may only have been for the nucleo-f103rb though.

Reply:
I think the problem is also there with F4 and the workaround could be, as you noticed, to initialize uart before i2c. That's what I did when debugging this driver some months ago.

Reply:
It would be nice for debugging to have stdio working as soon as possible during boot. But that is unrelated to this PR. Keeping those DEBUG statements will bite the next person trying to debug I2C unless someone would change the boot up process to initialize stdio sooner. So I agree that removing them is the better alternative.

Reply:
Noted, I opened an issue #10614 to document this concern.  I also did write that comment explaining it.